stub/src/main/java/io/grpc/kotlin/ServerCalls.kt

```diff
@@ -241,8 +239,14 @@ object ServerCalls {
     val rpcScope = CoroutineScope(context)
     rpcScope.async {
       val mutex = Mutex()
+      val headersSent = AtomicBoolean(false) // enforces only sending headers once
       val failure = runCatching {
         implementation(requests).collect {
```

Comment:
I think this breaks if the responses are empty.  Could we at least also do the check after collection is complete?

Reply:
Good catch, thanks. Added that and tests to catch that case.